### PR TITLE
MAT-6561: Alphabetize Test Case Drop Downs

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-6561](https://jira.cms.gov/browse/MAT-6561)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/common/codeInput/CodeInput.tsx [hunk 1]
```diff
@@ -115,13 +115,23 @@ const CodeInput = ({
...
-      return (
-        <MenuItem key={vs.oid} value={vs.oid} data-testid={`option-${vs.oid}`}>
-          {vs.display_name}
-        </MenuItem>
-      );
-    }),
+    valueSets
+      ?.sort(function (a, b) {
+        var nameA = a.display_name.toUpperCase();
+        var nameB = b.display_name.toUpperCase();
```
Comment (RohitKandimalla) [inline src/components/common/codeInput/CodeInput.tsx]:
```
_.sortBy(valueSets, ("display_name"))?.map((vs) => {
        return (
            <MenuItem key={vs.oid} value={vs.oid} data-testid={`option-${vs.oid}`}>
              {vs.display_name}
            </MenuItem>
        );
      })
```

We could use the help of lodash

File: src/components/common/codeInput/CodeInput.tsx [hunk 2]
```diff
@@ -115,13 +115,23 @@ const CodeInput = ({
...
+            value={vs.oid}
+            data-testid={`option-${vs.oid}`}
+          >
+            {vs.display_name}
+          </MenuItem>
+        );
+      }),
   ];
 
   const codeSystemMenuOptions = () => {
```
Comment (RohitKandimalla) [inline src/components/common/codeInput/CodeInput.tsx]: Why are we not sorting CodeSystem and code Menu options ?

Comment (RohitKandimalla) [reply src/components/common/codeInput/CodeInput.tsx]: Story did mention CodeSystem drop downs, but not codes, may be we can check with POs regarding codes

Comment (RohitKandimalla) [reply src/components/common/codeInput/CodeInput.tsx]: There are dropdowns from Codes, Screenshot from Story. I believe that will be `madie-patient/src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/Codes/Codes.tsx`
image

File: src/components/common/codeInput/CodeInput.tsx [hunk 3]
```diff
@@ -115,7 +116,7 @@ const CodeInput = ({
     >
       Custom Code
     </MenuItem>,
-    valueSets?.map((vs) => {
+    _.sortBy(valueSets, "display_name")?.map((vs) => {
```
Comment (RohitKandimalla) [inline src/components/common/codeInput/CodeInput.tsx:119]: We also need to sort `codeSystemMenuOptions` in this component

Comment (RohitKandimalla) [reply src/components/common/codeInput/CodeInput.tsx:119]: It felt like we need `codeMenuOptions` to also be sorted, but not sure if we need to sort them based on codes or based on display_name. Did we get a chance to check with POs ?

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/Codes/Codes.tsx [hunk 4]
```diff
@@ -118,15 +118,17 @@ const Codes = ({
...
-      ...Object.entries(codeSystems).map(([oid, name]) => (
-        <MenuItem
-          key={oid}
-          value={name as string}
-          data-testid={`code-system-option-${name}`}
-        >
-          {name}
-        </MenuItem>
-      )),
+      ...Object.entries(codeSystems)
```
Comment (RohitKandimalla) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/Codes/Codes.tsx:121]: Should `relatedCodeConcepts` be also sorted ?

Comment (ethankaplan) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/Codes/Codes.tsx:121]: It is sorted by codes by default, which the POs have said is good

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**